spec/operators/shareReplay-spec.ts

```diff
@@ -199,6 +199,19 @@ describe('shareReplay operator', () => {
     expectObservable(shared, sub2).toBe(expected2);
   });
 
+  it('should not restart due to unsubscriptions if refCount is true when the source has completed', () => {
+    const source = cold('a-(b|)         ');
+    const sub1 =        '^------!       ';
+    const expected1 =   'a-(b|)         ';
+    const sub2 =        '-----------^!  ';
+    const expected2 =   '-----------(b|)';
+
+    const shared = source.pipe(shareReplay({ bufferSize: 1, refCount: true }));
+
+    expectObservable(shared, sub1).toBe(expected1);
+    expectObservable(shared, sub2).toBe(expected2);
```

Comment:
We also need to test that there is only one subscription to the `source`. That can be done with `expectSubscriptions` which takes an expected subscription marble diagram - or an array of diagrams if multiple subscriptions are expected.

Something like this:

```ts
    const source     = cold('a-(b|)         ');
    const sourceSubs =      '^-!            ';
    /* ... */
    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
```

If there multiple subscriptions to the source are made, this expectation would fail, as `sourceSubs` would have to be an array of subscriptions.

Reply:
Yep. That makes a lot of sense. I've amended that commit with this expectation, thanks!

I've also added another commit that does the same thing on [other tests that were not doing that](https://github.com/ReactiveX/rxjs/pull/5457/commits/2863c9d792859dd01b68498afbe9c353dc32eaf0). I do understand that for those tests that expectation may not be as important, because the current expectations are "implicitly" checking for that already. So, I don't mind dropping this commit from the PR. I thought that maybe for consistency it could be worth it to have those expectations?